proto/prost/private/protoc_wrapper.rs

```diff
@@ -0,0 +1,1047 @@
+fn get_and_create_output_dir(out_dir: &Path, label: &str) -> PathBuf {
+    let out_dir = get_output_dir(out_dir, label);
+    if out_dir.exists() {
+        fs::remove_dir_all(&out_dir).expect("Failed to remove old output directory");
+    }
+    fs::create_dir_all(&out_dir).expect("Failed to create output directory");
+    out_dir
+}
+
+fn main() {
+    // Always enable backtraces for the protoc wrapper.
+    env::set_var("RUST_BACKTRACE", "1");
+
+    let Args {
+        protoc,
+        out_dir,
+        crate_name,
+        label,
+        package_info_file,
+        proto_files,
+        includes,
+        descriptor_set,
+        out_librs,
+        rustfmt,
+        proto_paths,
+        is_tonic,
+        extra_args,
+    } = Args::parse().expect("Failed to parse args");
+
+    let out_dir = get_and_create_output_dir(&out_dir, &label);
+
+    let mut cmd = process::Command::new(&protoc);
+    cmd.arg(format!("--prost_out={}", out_dir.display()));
+    if is_tonic {
+        cmd.arg(format!("--tonic_out={}", out_dir.display()));
+    }
+    cmd.args(extra_args);
+    cmd.args(
+        proto_paths
+            .iter()
+            .map(|proto_path| format!("--proto_path={}", proto_path)),
+    );
+    cmd.args(includes.iter().map(|include| format!("-I{}", include)));
+    cmd.args(&proto_files);
+
+    let status = cmd.status().expect("Failed to spawn protoc process");
+    if !status.success() {
+        panic!(
+            "protoc failed with status: {}",
+            status.code().expect("failed to get exit code")
+        );
+    }
+
+    // Not all proto files will consistently produce `.rs` or `.tonic.rs` files. This is
+    // caused by the proto file being transpiled not having an RPC service or other protos
+    // defined (a natural and expected situation). To guarantee consistent outputs, all
+    // `.rs` files are either renamed to `.tonic.rs` if there is no `.tonic.rs` or prepended
+    // to the existing `.tonic.rs`.
+    if is_tonic {
+        let tonic_files: BTreeSet<PathBuf> = find_generated_rust_files(&out_dir);
+
+        for tonic_file in tonic_files.iter() {
+            let tonic_path_str = tonic_file.to_str().expect("Failed to convert to str");
+            let filename = tonic_file
+                .file_name()
+                .expect("Failed to get file name")
+                .to_str()
+                .expect("Failed to convert to str");
+
+            let is_tonic_file = filename.ends_with(".tonic.rs");
+
+            if is_tonic_file {
+                let rs_file_str = format!(
+                    "{}.rs",
+                    tonic_path_str
+                        .strip_suffix(".tonic.rs")
+                        .expect("Failed to strip suffix.")
+                );
+                let rs_file = PathBuf::from(&rs_file_str);
+
+                if rs_file.exists() {
+                    let rs_file = rs_file.canonicalize().unwrap_or_else(|err| {
```

Comment:
Why do we need to canonicalize here? Do you really need to resolve symlinks? Need an absolute path? Or something else?

Reply:
I recall chasing down some missing files issues and this was the result of me trying to debug where the heck my files went. I just tested without this and I can remove it - thanks!